### PR TITLE
Support tsconfig base url

### PR DESCRIPTION
**Pull request by clinyong**

### CODE REVIEW

File: src/path-alias.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const files = await vscode.workspace.findFiles(include, exclude);
+
+  const mapping: PathAlias = {};
+  for (let i = 0; i < files.length; i++) {
+    try {
+      const fileContent = await fse.readFile(files[i].fsPath, "utf8");
+      const configFile = JSON5.parse(fileContent);
+      const baseUrl = getBaseUrlFromTsConfig(configFile);
+      if (baseUrl) {
+        mapping[baseUrl] = WORKSPACE_FOLDER_VARIABLE + "/" + baseUrl;
```
Comment (iChenLei) [inline src/path-alias.ts:59]: Not support paths property ?
> Yep, `paths property` is also **TODO** for PathIntellisense
https://github.com/ChristianKohler/PathIntellisense/blob/1103971e5eb8b147f8a76302e8861534fc79162b/src/configuration/tsconfig.service.ts#L49

File: src/path-alias.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    newAlias[key] = pathAlias[key].replace(WORKSPACE_FOLDER_VARIABLE, rootPath);
+  }
+
+  return newAlias;
+}
+
+const getTsAlias = memoize(async function (
+  workfolder: vscode.WorkspaceFolder
+): Promise<PathAlias> {
+  const include = new vscode.RelativePattern(workfolder, "[tj]sconfig.json");
```
[the remark this answers is not in the document]
Comment (iChenLei) [reply src/path-alias.ts]: ok, solved.

File: src/path-alias.ts (hunk 1 above)
Comment (iChenLei) [reply src/path-alias.ts]: tsconfig set paths alias config
```js
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@clinyong": ["./src/clinyong/*"]  // <- support this
    }
  }
}
```
css modules use paths alias
```jsx
import * as s from '@clinyong/button.module.scss';
```

Comment (clinyong) [reply src/path-alias.ts]: em... still working in progress...

Comment (clinyong) [reply src/path-alias.ts]: As we support the paths property, the `baseUrl` seems not necessary, so I remove it.

File: src/utils/ts-alias.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  Object.keys(paths).forEach((k) => {
+    formatPaths[removeEndSign(k)] = paths[k].map(removeEndSign);
+  });
+
+  return formatPaths;
+}
+
+export function _getAliasFromTsConfigPaths(tsconfig: {
```
Comment (clinyong) [inline src/utils/ts-alias.ts:54]: @iChenLei I don't know whether I cover all the tsconfig paths use cases. By now, I only remove the `* ` if exist in key or value, and join with baseUrl.

Comment (iChenLei) [reply src/utils/ts-alias.ts:54]: I think your test case is enough, very comprehensive consideration !
> src/test/suite/ts-alias.test.ts 👍🏻

**Comment by clinyong** [comment]
Can we merge this? @iChenLei

**Comment by iChenLei** [comment]
Yes, Thanks for great work.

**Pull request merged**